docs/trash.md

```diff
@@ -0,0 +1,97 @@
+Trash
+=====
+
+Under normal circumstances, when an item in Box is deleted, it is not actually erased immediately. Instead, it is
+moved to the Trash. The Trash allows you to recover files and folders that have been deleted. By default, items in
+the Trash will be purged after 30 days.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [List Trashed Items](#list-trashed-items)
+- [Get Trashed File, Folder or WebLink](#get-trashed-file-folder-or-weblink)
+- [Restore File, Folder, or WebLink from Trash](#restore-file-folder-or-weblink-from-trash)
+- [Permanently Delete File, Folder or WebLink](#permanently-delete-file-folder-or-weblink)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+List Trashed Items
+------------------
+
+To retrieve a list of all trashed items, you can use `client.trash().get_trashed_items(offset=None, limit=None, fields=None)`
+
+```python
+trashed_items = client.trash().get_trashed_items()
+for trashed_item in trashed_items:
+    # Do something
+```
+
+Get Trashed File, Folder or WebLink
+-----------------------------------
+
+To retrieve a file, folder, weblink from the trash, use `client.trash.get_from_trash(item, fields=None)`. 
+
+```python
+file = client.file('11111')
+file_from_trash = client.trash().get_from_trash(file)
+```
+
+```python
+folder = client.folder('22222')
+folder_from_trash = client.trash().get_from_trash(folder)
+```
+
+```python
+web_link = client.web_link('33333')
+web_link_from_Trash = client.trash().get_from_trash(web_link)
+```
+
+Restore File, Folder, or WebLink from Trash
+-------------------------------------------
+
+To retore a trashed item, effectively undeleting it, call `client.trash().restore_from_trash(item, name=None, parent_id=None, fields=None)`.
+
+```python
+file_to_restore = client.file('11111')
+restored_file = client.trash().restore_from_trash(file_to_restore)
+```
+
+```python
+folder_to_restore = client.folder('22222')
+restored_folder = client.trash().restore_from_trash(folder_to_restore)
+```
+
+```python
+web_link_to_restore = client.web_link('33333')
+restored_web_link = client.trash().restore_from_trash(web_link_to_restore)
+```
+
+In order to avoid conflicts, you can set a new name and new parent folder for the item you wish to restore.
+
+```python
+file = client.file('11111')
+new_name = 'New File Name'
+new_parent_id = '1111'
+restored_file = client.trash().restore_from_trash(file, new_name, new_parent_id)
+```
+
+Permanently Delete File, Folder or WebLink
+-------------------------------------------
+
+To delete an item from trash, use `client.trash().permanently_delete_item(item)`.
+
+```python
+file = client.file('11111')
+client.trash().permanently_delete_item(file)
```

Comment:
`file` is a builtin type in Python 2.7; consider using a different variable name here